processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +355,154 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
+            futures.add(
+                executorService.submit(
+                    () -> {
+
+                      LookupExtractorFactoryContainer container = lookupBean.getContainer();
+                      if (container.getLookupExtractorFactory().start()) {
+                        LOG.info(
+                            "Started lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+                      } else {
+                        LOG.error(
+                            "Failed to start lookup [%s]:[%s]",
+                            lookupBean.getName(),
+                            container
+                        );
+                        return null;
+                      }
+                    }
+                )
+            );
+          }
+          final ListenableFuture<List<Map.Entry>> futureList = Futures.allAsList(futures);
+          try {
+            futureList.get()
+                      .stream()
+                      .filter(Objects::nonNull)
+                      .forEach(builder::put);
+          }
+          catch (Exception ex) {
+            futureList.cancel(true);
+            throw ex;
+          }
+          stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
+        }
+        catch (Exception e) {
+          LOG.error(e, "Failed to finish lookup load process.");
+        }
+        executorService.shutdownNow();
+      } else {
+        LOG.info("No lookups to be loaded at this point");
+        stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+      }
+    } else {
+      LOG.info("Lookup synchronization is disabled");
+      stateRef.set(new LookupUpdateState(ImmutableMap.of(), ImmutableList.of(), ImmutableList.of()));
+    }
+  }
 
-        if (container.getLookupExtractorFactory().start()) {
-          builder.put(lookupBean.getName(), container);
+  private List<LookupBean> getLookupListFromCoordinator(String tier)
+  {
+    try {
+      final StatusResponseHolder response = fetchLookupsForTier(tier);
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      if (!response.getStatus().equals(HttpResponseStatus.OK)) {
+        LOG.error(
+            "Error while fetching lookup code from Coordinator with status[%s] and content[%s]",
+            response.getStatus(),
+            response.getContent()
+        );
+        if (lookupSnapshotTaker != null) {
+          LOG.info("Attempting to load saved snapshot");
+          return null;
+        }
+      } else {
+        // Older version of getSpecificTier returns a list of lookup names.
+        // Lookup loading is performed via snapshot if older version is present.
+        if (response.getContent().startsWith("[")) {
```

Comment:
can you update comment to say, this check should be removed in a future release?